Source/Core/VideoBackends/Vulkan/CMakeLists.txt

```diff
@@ -1,3 +1,27 @@
+# Enable Vulkan by default everywhere except macOS, as it's not supported there.
+# It can be compiled, however there is no WSI, so it can only run headless.
+if(NOT APPLE)
+  set(ENABLE_VULKAN_DEFAULT ON)
+else()
+  set(ENABLE_VULKAN_DEFAULT OFF)
+endif()
+option(ENABLE_VULKAN "Enables Vulkan backend" ${ENABLE_VULKAN_DEFAULT})
+
+if(NOT ENABLE_VULKAN)
+  return()
+endif()
+
+# We require libX11-xcb when building for X11. This isn't required for any other backend, so test for it here.
+if(USE_X11)
+  find_package(X11_XCB)
+  if (NOT X11_XCB_FOUND)
+    message(WARNING "libX11-xcb is required to build the Vulkan backend with X11. Disabling Vulkan backend.")
```

Comment:
I'd tend to agree. However, I discussed this with @Orphis and IIRC they preferred disabling the backend in this case. I don't mind either way.